sync/tc.py

```diff
@@ -34,12 +37,37 @@
 env = Environment()
 
 
+tc_base_cache = {}
+
+
+def get_tc_base(task_id):
+    defaults = (TASKCLUSTER_ROOT_URL, QUEUE_BASE, INDEX_BASE)
+    old = (OLD_TASKCLUSTER_ROOT_URL, OLD_QUEUE_BASE, OLD_INDEX_BASE)
+    if task_id not in tc_base_cache:
+        try:
+            resp = requests.get(QUEUE_BASE + "task/%s" % task_id)
+            resp.raise_for_status()
+        except requests.HTTPError:
+            try:
+                resp = requests.get(OLD_QUEUE_BASE + "task/%s" % task_id)
+                resp.raise_for_status()
+            except requests.HTTPError:
+                value = defaults
```

Comment:
Is this the right action to take here? I realise this is very edge case but if the tc base is the old one but there was a network issue, then this will get cached as being on the new taskcluster.

Reply:
OK, updated.